server/spec/refresh_pools_spec.rb

```diff
@@ -214,6 +214,154 @@
     expect(consumer.entitlementCount).to eq(0)
   end
 
+  it 'revokes entitlements when pool quantity changes' do
+    owner_key = random_string('test_owner')
+    owner = create_owner(owner_key)
+
+    product = create_upstream_product(random_string('test_prod'), {
+      :attributes => {
+        'multi-entitlement' => 'yes'
+      }
+    })
+
+    sub = create_upstream_subscription(random_string('test_sub'), owner_key, { :quantity => 6,
+      :product => product })
+
+    @cp.refresh_pools(owner_key)
+    pools = @cp.list_pools({:owner => owner.id})
+    expect(pools.length).to eq(1)
+
+    user = user_client(owner, random_string('test_user'))
+    consumer1 = consumer_client(user, random_string('test_consumer'))
+    consumer2 = consumer_client(user, random_string('test_consumer'))
+
+    entitlements = consumer1.consume_pool(pools.first['id'], {:quantity => 2})
+    expect(entitlements.length).to eq(1)
+    entitlements = consumer2.consume_pool(pools.first['id'], {:quantity => 4})
+    expect(entitlements.length).to eq(1)
+
+    consumer_entitlements = @cp.list_entitlements({:uuid => consumer1.uuid})
+    expect(consumer_entitlements.length).to eq(1)
+
+    consumer_entitlements = @cp.list_entitlements({:uuid => consumer2.uuid})
+    expect(consumer_entitlements.length).to eq(1)
+
+    # Update subscription, reducing its quantity, then refresh. The entitlements should be removed.
+    update_upstream_subscription(sub.id, { :quantity => 3 })
+    @cp.refresh_pools(owner_key)
+
+    consumer_entitlements = @cp.list_entitlements({:uuid => consumer1.uuid})
+    expect(consumer_entitlements.length).to eq(1)
+
+    consumer_entitlements = @cp.list_entitlements({:uuid => consumer2.uuid})
+    expect(consumer_entitlements.length).to eq(0)
+  end
+
+  it 'revokes entitlements when pool start date changes' do
+    owner_key = random_string('test_owner')
```

Comment:
These first 3 tests do not really test something significantly different from each other (quantity, end date, start date), so I'd say we only need 1 of them?

Reply:
Given the lack of details in the BZ, I was making sure that every change that resulted in entitlement revocation was covered. Yes, there's overlap, but that's the nature of testing each path.